### ISSUE TITLE
UNIQUE constraint failed: auth_user.email

### ISSUE THREAD

**Issue by marchss** [pennersr/django-allauth#1152]
I am having this wired error after the integration with django-allauth successfully running for several days. The email is used here as unique identifier of user instead of the user_name

IntegrityError at /accounts/signup/
UNIQUE constraint failed: auth_user.email
Request Method:	POST
Request URL:	http://localhost:8080/accounts/signup/
Django Version:	1.8
Exception Type:	IntegrityError
Exception Value:	
UNIQUE constraint failed: auth_user.email
Exception Location:	~/web_dev/unicorn_dev/lib/python3.4/site-packages/django/db/backends/sqlite3/base.py in execute, line 318
Python Executable:	~/web_dev/unicorn_dev/bin/python
Python Version:	3.4.3
Python Path:	
['~/web_dev/unicorn_dev/funding',
 '/Library/Frameworks/Python.framework/Versions/3.4/lib/python34.zip',
 '/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4',
 '/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/plat-darwin',
 '/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/lib-dynload',
 '~/web_dev/unicorn_dev/lib/python3.4/site-packages']

Here is the Model class:

class AccountUser(AbstractBaseUser, PermissionsMixin):
    email = models.EmailField(_('email address'), blank=False, max_length=255, unique=True)
    first_name = models.CharField(_('first name'), max_length=40, blank=True, null=True, unique=False)
    last_name = models.CharField(_('last name'), max_length=40, blank=True, null=True, unique=False)

Here is the settings.py
ACCOUNT_USER_MODEL_USERNAME_FIELD = None  /* user_name is not used, if not None, migration will report errors */
ACCOUNT_USERNAME_REQUIRED = False
#Enforce uniqueness of e-mail addresses.
ACCOUNT_USER_MODEL_EMAIL_FIELD = "email"
ACCOUNT_UNIQUE_EMAIL = True
ACCOUNT_AUTHENTICATION_METHOD = "email" 

Could anyone please help?

**Comment by marchss** [comment]
I think this one is different from the https://github.com/pennersr/django-allauth/issues/1014

**Comment by pennersr** [comment]
I do not think there is an issue in allauth here. All sorts of weirdness can occur if you DB is not configured properly, or, if e.g. the allauth uniqueness settings were changed somewhere along the 
way.

For example, have a look at this example I bumped into while debugging an installation with similar complaints:

    >>> EmailAddress.objects.filter(email='info@FooBar.com') 	
    [<EmailAddress: info@FooBar.com  (foobar)>]  		

    >>> EmailAddress.objects.filter(email__iexact='info@FooBar.com') 		
    []

So, when searching using __iexact not results were found. That is clearly wrong.

https://docs.djangoproject.com/en/dev/ref/databases/#mysql-collation
